lib/gpprmsya.gi

```diff
@@ -329,36 +329,44 @@ local   F,      # free group
     mov:=MovedPoints(G);
     deg:=Length(mov);
 
-    # create the finitely presented group with <G>.degree-1 generators
-    F := FreeGroup( 2, str);
-    gens:=GeneratorsOfGroup(F);
+    #special case for degree 3, cyclic
+    if deg=3 then
+      F := FreeGroup( 1, str);
+      imgs:=[(1,2,3)];
```

Comment:
That's wrong in general. E.g.
```
gap> IsomorphismFpGroup(Group((4,7,3)));
[ (1,2,3) ] -> [ F1 ]
```

How about this alternative fix?
```patch
diff --git a/lib/gpprmsya.gi b/lib/gpprmsya.gi
index e53fb5ee6..57c91e158 100644
--- a/lib/gpprmsya.gi
+++ b/lib/gpprmsya.gi
@@ -329,6 +329,9 @@ local   F,      # free group
     mov:=MovedPoints(G);
     deg:=Length(mov);
 
+    # special case for degree 3, cyclic
+    if deg = 3 then TryNextMethod(); fi;
+
     # create the finitely presented group with <G>.degree-1 generators
     F := FreeGroup( 2, str);
     gens:=GeneratorsOfGroup(F);
```

Reply:
I submitted https://github.com/gap-system/gap/pull/3647 as an alternative fix along these lines, and also with a test case added.

Reply:
Thanks for spotting this. I have fixd it (by writing down the permutation explicitly).